sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ApplyDefaultCollationToStringType.scala

```diff
@@ -82,6 +89,55 @@ object ApplyDefaultCollationToStringType extends Rule[LogicalPlan] {
     }
   }
 
+  /**
+   * Fetches the default collation specified at the schema level in the given logical plan.
+   * If no schema-level collation is specified, returns None. We don't match for non-DDL commands
+   * because schema/catalog default collation affects objects only during their creation.
+   */
+  private def fetchSchemaLevelDefaultCollation(plan: LogicalPlan): Option[String] = {
+    try {
+      plan match {
+        case CreateTable(ResolvedIdentifier(catalog: SupportsNamespaces, identifier), _, _, _, _) =>
```

Comment:
to simplify the code, I think we only need to preprocess CreateTable/ReplaceTable by specify the collation if it's not present and the schema defines one. Then the code can be same as before: `fetchDefaultCollation` and then `transform`